aten/src/ATen/native/TensorShape.cpp

```diff
@@ -909,9 +909,27 @@ Tensor view(const Tensor& self, IntArrayRef size) {
     "not compatible with input tensor's size and stride (at least one dimension"
     " spans across two contiguous subspaces). Use .reshape(...) instead.");
   auto stride_value = *stride;
-  auto self_ = self.clone();
-  self_.set_(self.storage(), self.storage_offset(), inferred_size,
-             stride_value);
+  auto self_ = self.alias();
+  self_.set_(
+    self.storage(), self.storage_offset(), inferred_size, stride_value);
+  return self_;
+}
+
+Tensor alias(const Tensor& self) {
+  Tensor self_;
+  if (self.is_quantized()) {
+    TORCH_CHECK(self.qscheme() == kPerTensorAffine,
+                "Only PerTensorAffine quantization is supported right now");
+    auto impl = c10::make_intrusive<QTensorImpl>(
+                    Storage(self.storage()),
+                    self.type_id(),
+                    get_qtensorimpl(self)->quantizer());
+    self_ = Tensor(impl);
+  } else {
+    auto impl = c10::make_intrusive<TensorImpl>(Storage(self.storage()),
```

Comment:
this doesn't look right -- in the call you are replacing, the sizes and strides are set correctly.  This doesn't matter for `view` because you never look at the alias before you reset it, but this is wrong for the other users of alias.

Reply:
oh right.